isthmus/src/test/java/io/substrait/isthmus/ExtendedExpressionTestBase.java

```diff
@@ -0,0 +1,51 @@
+package io.substrait.isthmus;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import io.substrait.proto.ExtendedExpression;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.calcite.sql.parser.SqlParseException;
+
+public class ExtendedExpressionTestBase {
+  public static String asString(String resource) throws IOException {
+    return Resources.toString(Resources.getResource(resource), Charsets.UTF_8);
+  }
+
+  public static List<String> tpchSchemaCreateStatements() throws IOException {
+    String[] values = asString("tpch/schema.sql").split(";");
+    return Arrays.stream(values)
+        .filter(t -> !t.trim().isBlank())
+        .collect(java.util.stream.Collectors.toList());
+  }
+
+  protected ExtendedExpression assertProtoExtendedExpressionRoundrip(String query)
+      throws IOException, SqlParseException {
+    return assertProtoExtendedExpressionRoundrip(query, new SqlToSubstrait());
+  }
+
+  protected ExtendedExpression assertProtoExtendedExpressionRoundrip(String query, SqlToSubstrait s)
+      throws IOException, SqlParseException {
+    return assertProtoExtendedExpressionRoundrip(query, s, tpchSchemaCreateStatements());
+  }
+
+  protected ExtendedExpression assertProtoExtendedExpressionRoundrip(
+      String query, SqlToSubstrait s, List<String> creates) throws SqlParseException {
+    io.substrait.proto.ExtendedExpression protoExtendedExpression =
+        s.executeSQLExpression(query, creates);
+
+    try {
+      String ee = JsonFormat.printer().print(protoExtendedExpression);
+      System.out.println("Proto Extended Expression: \n" + ee);
+
+      // FIXME! Implement test validation as the same as proto Plan implementation
```

Comment:
I assume this is probably a todo?

Reply:
Just implemented at https://github.com/substrait-io/substrait-java/pull/206